### ISSUE TITLE
unusedArguments fails when an argument is reassigned to a for loop's parameter

### ISSUE THREAD

**Issue by nicorichard** [nicklockwood/SwiftFormat#1184]
A minimally reproducible example for `0.49.8`

```swift
func foo(index: Int) {
    for index in 0..<1 {}
    print(index)
}
```

will be formatted to

```swift
func foo(index _: Int) {
    for index in 0..<1 {}
    print(index) // compile error
}
```

This seems to be a regression which may have been caused by the solution for my previous issue #1174

The issue does not occur for version `0.49.6`

---

Another example of a failing case in the same category:

```swift
func foo(index: Int) {
    for (index, value) in [].enumerated() {}
    print(index)
}
```

**Comment by nicklockwood** [comment]
Seems like a duplicate of https://github.com/nicklockwood/SwiftFormat/issues/1183 - I've landed a fix in develop already.

**Comment by nicklockwood** [comment]
@nicorichard fixed in 0.49.9
